### ISSUE TITLE
Swagger spec to include operationId ?

### ISSUE THREAD

**Issue by MarkEdmondson1234** [rstudio/plumber#154]
Hi, I used the swagger.json from the Plumber app to generate a Google Cloud Endpoint which is neat, but I had to add a `operationId` to the swagger spec to get it to work.  Here is my horrible hack:

```r
library(yaml)
library(jsonlite)

make_openapi <- function(projectId){

  ## this calls the Plumber apps swagger.json
  json <- jsonlite::fromJSON(sprintf("https://%s.appspot.com/swagger.json", projectId))

  ## change host from 127.0.0.1 to the appengine project
  json$host <- sprintf("%s.appspot.com", projectId)
  
  ## add operationId to each endpoint
  ohgod <- lapply(names(json$paths), 
                  function(x) {lapply(json$paths[[x]], 
                                      function(verb) {verb$operationId <- basename(x);verb})})
  json$paths <- setNames(ohgod, names(json$paths))

  # silly formatting to make these entries arrays and not characters
  yaml <- gsub("application/json", "[application/json]", yaml::as.yaml(json))
  yaml <- gsub("schemes: http", "schemes: [http]", yaml)
  
  writeLines(yaml, con = "openapi.yaml")
}

make_openapi("your-project-id")
```

**Comment by trestletech** [comment]
Thanks for the suggestion. Do you have any suggestions for how this field should be populated in the Swagger doc? I could just concatenate the verb with the path like `@get /my/path` becomes `operationId: get-my-path`? It gets a little awkward with endpoints that support multiple verbs, but I think the idea would still work.

Also would it be useful to include a `plumber.yaml` for your usecase?

**Comment by MarkEdmondson1234** [comment]
`operationId: get-my-path` would be perfect, I forgot to add the verb in my example.  It just needs to be unique.

My use case supports JSON or YAML files, but to modify it (such as the hostname) I turned it into YAML so a user can edit the final result easier.  Its not a big deal to convert between them.

**Comment by MarkEdmondson1234** [comment]
Actually what would be most useful would be if the `swagger.json` or yaml file could be generated just from the code, without needing to launch the plumber server.

**Comment by alexhallam** [comment]
Hi, I wanted to see if there was an update on this issue?

**Comment by schloerke** [comment]
@alexhallam Coming soon! (<= a few weeks)

**Comment by schloerke** [comment]
This can be solved in plumber v1.0.0 using

```
pr %>%
  pr_set_api_spec(function(spec) {
    # update spec object here
    # return spec object
    spec
  })
```
